load-balancer/least-requests/src/test/java/io/smallrye/stork/loadbalancer/leastresponsetime/LeastRequestsLoadBalancerTest.java

```diff
@@ -0,0 +1,171 @@
+        TestConfigProvider.addServiceConfig("first-service", "least-requests", "static",
+                null,
+                Map.of("address-list", String.format("%s,%s", FST_SRVC_1, FST_SRVC_2)));
+        TestConfigProvider.addServiceConfig("without-instances", "least-requests",
+                "empty-services", null, Collections.emptyMap());
+
+        stork = StorkTestUtils.getNewStorkInstance();
+    }
+
+    @Test
+    public void shouldSelectLessLoaded() {
+        Service service = stork.getService("first-service");
+
+        ServiceInstance service1, service2;
+
+        // First selection returns the first service
+        ServiceInstance instance = selectInstance(service);
+        assertThat(asString(instance)).isEqualTo(FST_SRVC_1);
+        assertThat(instance.gatherStatistics()).isTrue();
+        service1 = instance;
+
+        // Second selection returns the second service as there is an inflight-call in the first one
+        instance = selectInstance(service);
+        assertThat(asString(instance)).isEqualTo(FST_SRVC_2);
+        assertThat(instance.gatherStatistics()).isTrue();
+        service2 = instance;
+
+        // Report termination of the service 2 call
+        instance.recordResult(10, null);
+
+        // Next selection still return the second service, as we still have an inflight-call in service 1
+        instance = selectInstance(service);
+        assertThat(asString(instance)).isEqualTo(FST_SRVC_2);
+        assertThat(instance.gatherStatistics()).isTrue();
+
+        service1.recordResult(1000, null);
+        // Now select 1 as they have the same amount of inflight
+        instance = selectInstance(service);
+        assertThat(asString(instance)).isEqualTo(FST_SRVC_1);
+        assertThat(instance.gatherStatistics()).isTrue();
+    }
+
+    @Test
+    public void shouldSelectLessLoadedWhenBothLoaded() {
+        Service service = stork.getService("first-service");
+
+        ServiceInstance service1, service2;
+
+        // First selection returns the first service
+        ServiceInstance instance = selectInstance(service);
+        assertThat(asString(instance)).isEqualTo(FST_SRVC_1);
+        assertThat(instance.gatherStatistics()).isTrue();
+        service1 = instance;
+
+        // Second selection returns the second service as there is an inflight-call in the first one
+        instance = selectInstance(service);
+        assertThat(asString(instance)).isEqualTo(FST_SRVC_2);
+        assertThat(instance.gatherStatistics()).isTrue();
+        service2 = instance;
+
+        ServiceInstance last = service2;
+        for (int i = 0; i < 100; i++) {
+            ServiceInstance selected = selectInstance(service);
+            assertThat(asString(selected)).isNotEqualTo(asString(last));
+            last = selected;
+        }
+
+        Random random = new Random();
+
+        last = service1;
+        // Start reporting
+        for (int i = 0; i < 102; i++) {
```

Comment:
won't the last service have -2 results after this loop?

Reply:
Oops! forgot to change the upper bound.